flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ResultStore.java

```diff
@@ -61,7 +60,7 @@ public ResultStore(Configuration flinkConfig) {
 	 */
 	public <T> DynamicResult<T> createResult(Environment env, TableSchema schema, ExecutionConfig config) {
 
-		final TypeInformation<Row> outputType = Types.ROW_NAMED(schema.getFieldNames(), schema.getFieldTypes());
+		final RowTypeInfo outputType = (RowTypeInfo) Types.ROW_NAMED(schema.getFieldNames(), schema.getFieldTypes());
```

Comment:
Maybe let's just replace it with the `RowTypeInfo` ctor, if we expect that type anyway?